### PR TITLE
Added details for optional parameters

### PR DESCRIPTION
**Pull request by yoshizzle**
For some optional parameters, additional information is
helpful for users to understand what is required when other
attributes are set, what values are needed, etc.

### CODE REVIEW

**Comment by Dorthu** [comment]
Can you link the related issue?

**Comment by yoshizzle** [comment]
https://github.com/linode/developers/issues/54

File: _data/endpoints/linodes.yaml [hunk 1]
```diff
@@ -61,6 +62,19 @@ endpoints:
             optional: true
             description: The group to assign this Linode.
             limit: 0-50 characters
+              <ul><li>root_pass is required if the source provided is a distribution.</li></ul>
```
Comment (Dorthu) [inline _data/endpoints/linodes.yaml]: This should be on the root_pass attribute

**Comment by Dorthu** [comment]
Please add these:

POST /linodes:
 * label - defaults to "linode<id>"
 * group - defaults to empty
 * stackscript_udf_responses - defaults to {}, must match UDFs required by stackscript

POST /linodes/:id/disks
 * stackscript_udf_responses - defaults to {}, must match UDFs required by stackscript

Everything else looks good to me.

**Pull request merged**